mobly/controllers/android_device_lib/services/logcat.py

```diff
@@ -292,7 +292,7 @@ def _start(self):
             logcat_file_path = os.path.join(self._ad.log_path, f_name)
             self.adb_logcat_file_path = logcat_file_path
         utils.create_dir(os.path.dirname(self.adb_logcat_file_path))
-        cmd = '"%s" -s %s logcat -v threadtime -T 1 %s >> "%s"' % (
+        cmd = ' "%s" -s %s logcat -v threadtime -T 1 %s >> "%s" ' % (
```

Comment:
so the only change is adding extra spaces?
does this actually solve your problem?

Reply:
Yes! Only change is adding extra spaces!
My explanation copy from top of this thread:

> I trace the codes of Popen API and found there is a "patch_args" in pydev/.../pydev_monkey.py.
> https://github.com/JetBrains/intellij-community/blob/master/python/helpers/pydev/_pydev_bundle/pydev_monkey.py
> This function remove double quote in args if arg string start and end with double quote.
> Therefore, I provide this solution add space at beginning and at last,
> ' "%s" -s %s logcat -v threadtime -T 1 %s >> "%s" '
> this break the "patch_args" rule and can still run correctly on Windows, Linux and MacOS.

Reply:
This solution is really simple and actually solve IntelijIDEA debugging issue.

Reply:
Could add an inline comment noting the extra spaces are required?

Reply:
Could you add an inline comment noting the extra spaces are required?

Reply:
Thank you for your suggestion.
I added inline comment.
Thank you.